Marlin/status_screen_lite_ST7920.h

```diff
@@ -877,7 +877,7 @@ void ST7920_Lite_Status_Screen::update_status_or_position(bool forceUpdate) {
 
 void ST7920_Lite_Status_Screen::update_progress(const bool forceUpdate) {
   #if DISABLED(LCD_SET_PROGRESS_MANUALLY)
-    uint8_t progress_bar_percent = 0;
+    uint8_t progress_bar_percent = card.percentDone();
```

Comment:
@CryToCry96 — Try building with `SDSUPPORT` disabled and see how this fares.

CC @marcio-ao

Reply:
Note that the code below is already supposed to be setting the percentage from the card…
```cpp
  // Set current percentage from SD when actively printing
   #if ENABLED(LCD_SET_PROGRESS_MANUALLY) && ENABLED(SDSUPPORT) && (ENABLED(LCD_PROGRESS_BAR) || ENABLED(DOGLCD))
     if (IS_SD_PRINTING) progress_bar_percent = card.percentDone();
   #endif
```
One thing we know for sure at this point in the code is that `DOGLCD` has to be enabled. So the last clause can be removed:
```cpp
  // Set current percentage from SD when actively printing
   #if ENABLED(LCD_SET_PROGRESS_MANUALLY) && ENABLED(SDSUPPORT)
     if (IS_SD_PRINTING) progress_bar_percent = card.percentDone();
   #endif
```
The complete solution simply accounts for the fact that `progress_bar_percent` doesn't exist yet if `LCD_SET_PROGRESS_MANUALLY` is disabled, so we're just making a local version in that case… Hence, the test for `ENABLED(LCD_SET_PROGRESS_MANUALLY)` before setting percentage from the SD card is the real bug. So the correct answer is to simply change the code below to this:
```cpp
  // Set current percentage from SD when actively printing
   #if ENABLED(SDSUPPORT)
     if (IS_SD_PRINTING) progress_bar_percent = card.percentDone();
   #endif
```

Reply:
And, last note… If neither `SDSUPPORT` nor `LCD_SET_PROGRESS_MANUALLY` are enabled, then no progress bar should be drawn at all.